client/state/notices/reducer.js

```diff
@@ -9,28 +9,29 @@ import { combineReducers } from 'redux';
 import {
 	NEW_NOTICE,
 	REMOVE_NOTICE,
-	SET_ROUTE
+	SET_ROUTE,
+	SERIALIZE,
+	DESERIALIZE
 } from 'state/action-types';
 
 export function items( state = [], action ) {
 	switch ( action.type ) {
 		case NEW_NOTICE:
-			state = [ action.notice, ...state ];
-			break;
+			return [ action.notice, ...state ];
```

Comment:
Our [javascript guidelines for switch statements](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#switch-statements) discourage returning from within a case statement. That's in a few places in this PR (and our codebase in general). I'm not sure where that came from or the original reasoning. Curious if anyone wants to champion the guideline or if we should just simplify our javascript guidelines by removing it.